benchmarks/src/commonMain/kotlin/BufferOps.kt

```diff
@@ -0,0 +1,331 @@
+            builder.append(part)
+        }
+        builder.setLength(length)
+        return builder.toString()
+    }
+
+    override fun padding(): ByteArray {
+        val baseString = constructString()
+        if (baseString.utf8Size() >= minGap) {
+            return baseString.encodeToByteArray()
+        }
+        val builder = StringBuilder((minGap * 1.5).toInt())
+        while (builder.length < minGap) {
+            builder.append(baseString)
+        }
+        return builder.toString().encodeToByteArray()
+    }
+
+    @Setup
+    fun setupString() {
+        string = constructString()
+    }
+
+    @Benchmark
+    fun benchmark(): String {
+        val s = buffer.size
+        buffer.writeUtf8(string)
+        return buffer.readUtf8(buffer.size - s)
+    }
+}
+
+open class Utf8LineBenchmarkBase: BufferRWBenchmarkBase() {
+    @Param("17")
+    var length: Int = 0
+
+    @Param("LF", "CRLF")
+    var separator: String = ""
+
+    protected var string: String = ""
+
+    private fun lineSeparator(): String = when (separator) {
+        "LF" -> "\n"
+        "CRLF" -> "\r\n"
+        else -> throw IllegalArgumentException("Unsupported line separator type: $separator")
+    }
+
+    private fun constructString(): String = ".".repeat(length) + lineSeparator()
+
+    override fun padding(): ByteArray {
+        val string = constructString()
+        if (string.length >= minGap) {
+            return string.encodeToByteArray()
+        }
+        val builder = StringBuilder((minGap * 1.5).toInt())
+        while (builder.length < minGap) {
+            builder.append(string)
+        }
+        return builder.toString().encodeToByteArray()
+    }
+
+    @Setup
+    fun setupString() {
+        string = constructString()
+    }
+}
+
+open class Utf8LineBenchmark: Utf8LineBenchmarkBase() {
+    @Benchmark
+    fun benchmark(): String? {
+        buffer.writeUtf8(string)
+        return buffer.readUtf8Line()
+    }
+}
+
+open class Utf8LineStrictBenchmark: Utf8LineBenchmarkBase() {
+    @Benchmark
+    fun benchmark(): String {
+        buffer.writeUtf8(string)
+        return buffer.readUtf8LineStrict()
+    }
+}
+
+private const val INDEX_OF_TARGET_VAL: Byte = 1
+
+@State(Scope.Benchmark)
+open class IndexOfBenchmark {
+
+    @Param("128")
+    var dataSize: Int = 0
+
+    @Param("-1", "7", "100")
+    var valueOffset: Int = 0
+
+    @Param("true", "false")
+    var spanOverMultipleSegment: Boolean = false
+
+    private val buffer = Buffer()
+
+    @Setup
+    fun setupBuffers() {
+        val array = ByteArray(dataSize)
+        if (valueOffset >= 0) array[valueOffset] = INDEX_OF_TARGET_VAL
+
+        var paddingSize = 0L
+        if (spanOverMultipleSegment) {
+            paddingSize = 8192L - dataSize / 2L
+        }
+        val padding = ByteArray(paddingSize.toInt())
+        buffer.write(padding).write(array).skip(paddingSize)
+    }
+
+    @Benchmark
+    fun benchmark(): Long = buffer.indexOf(INDEX_OF_TARGET_VAL)
```

Comment:
Do I understand correctly that this test involves searching for a value only within one segment?